### DISCUSSION TITLE
how to make rotate class to work with js click method?

### DISCUSSION THREAD

**Discussion by CatPeare** [tailwindlabs/tailwindcss#11145] [Help]
I am using group-hover already, but I want rotate class to work when button is clicked (I will not use group-hover, it is already removed from the code but used in the example to show what I did before). even if I add rotate class when button is clicked; it adds the class, but class doesn't work.
```
        <div>
            <button class="group bg-stone-700 rounded-2xl">
                <svg class="fill-emerald-400 group-hover:rotate-[1080deg] transition-all duration-700" width="36"
                    height="36" fill="#FFF" viewBox="0 0 24 24">
                    <path fill="none" d="M0 0h24v24H0z" />
                    <path d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6z" />
                </svg>
            </button>
        </div>
```

**Comment by wongjn** [comment]
Assuming the `rotate-*` class you add on button click is one without a variant, this class would always be overridden by the `group-hover:rotate-[1080deg]` class since `group-hover:rotate-[1080deg]` would have higher specificity. Consider using [the important modifier](https://tailwindcss.com/docs/configuration#important-modifier) to add `!important` to the `rotate-*` class you add on button click, like `!rotate-90`.

**Reply by CatPeare** [reply]
> Assuming the `rotate-*` class you add on button click is one without a variant, this class would always be overridden by the `group-hover:rotate-[1080deg]` class since `group-hover:rotate-[1080deg]` would have higher specificity. Consider using [the important modifier](https://tailwindcss.com/docs/configuration#important-modifier) to add `!important` to the `rotate-*` class you add on button click, like `!rotate-90`.


I think I expressed myself improperly sorry for that.
I don't want to use `group-hover:rotate-[1080deg]`  in actual code; but I want to add this rotate effect when button is clicked. when creating an event listener, the class is added when button is clicked. but it doesn't rotate.

**Reply by wongjn** [reply]
Consider checking:
- The file that contains the JavaScript code is covered by the `content` file globs.
- The class name string appears in a file as a full unbroken string, i.e. not built via string concatenation/interpolation.

**Reply by CatPeare** [reply]
I have made it work, but the thing I don't understand it didn't work with id selector. once I changed it to query it worked. thank you by the way for your time. here is the part of solution (typescript)
```
const rotate = document.querySelector("#rotate");
buttonToTop?.addEventListener("click", () => {
    rotate?.classList.add("rotate-90");
});
```
